### PR TITLE
feat: LT-1332 indicate a location type on an SICL

### PR DESCRIPTION
**Pull request by rorysproule1**

### CODE REVIEW

File: applications/views/external_locations.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ def _set_locations_and_sites(
 
             # Transhipment and Trade Control applications can't have sites based in certain countries
             if application.case_type.id in [*CaseTypeEnum.trade_control_case_type_ids(), CaseTypeEnum.SITL.id]:
-                if new_location.country.id in TRANSHIPMENT_AND_TRADE_CONTROL_BANNED_COUNTRIES:
+                if new_location.country and new_location.country.id in TRANSHIPMENT_AND_TRADE_CONTROL_BANNED_COUNTRIES:
```
[the remark this answers is not in the document]
Comment (rorysproule1) [reply applications/views/external_locations.py:101]: Amended to just 

`if new_location.country`

File: organisations/serializers.py [hunk 2]
```diff
@@ -275,14 +276,34 @@ class Meta:
...
+    address = serializers.CharField()
+    country = CountrySerializerField(required=False)
+    organisation = serializers.PrimaryKeyRelatedField(queryset=Organisation.objects.all())
+    location_type = serializers.CharField(
+        required=True, error_messages={"required": strings.ExternalLocations.Errors.LOCATION_TYPE},
+    )
+
+    def validate(self, data):
+        if data["location_type"] == "land_based":
+            if not data["country"]:
```
[the remark this answers is not in the document]
Comment (rorysproule1) [reply organisations/serializers.py:298]: Amended to

`if data["location_type"] == "land_based" and not data["country"]:`

File: organisations/tests/tests_external_locations.py [hunk 3]
```diff
@@ -33,3 +33,53 @@ def test_failed_create_external_location(self):
...
+        data = {
+            "name": "regional site",
+            "address": "123 Test",
+            "location_type": "sea_based",
+            "application_type": "sicl",
+        }
+
+        response = self.client.post(self.url, data, **self.exporter_headers)
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```
[the remark this answers is not in the document]
Comment (rorysproule1) [reply organisations/tests/tests_external_locations.py:67]: Done

File: organisations/tests/tests_external_locations.py [hunk 4]
```diff
@@ -33,3 +33,53 @@ def test_failed_create_external_location(self):
...
+            "address": "123 Test",
+            "country": "FR",
+            "location_type": "",
+            "application_type": "sicl",
+        }
+
+        response = self.client.post(self.url, data, **self.exporter_headers)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(ExternalLocation.objects.all().count(), 1)
```
[the remark this answers is not in the document]
Comment (rorysproule1) [reply organisations/tests/tests_external_locations.py:85]: Done!

File: organisations/tests/tests_external_locations.py [hunk 5]
```diff
@@ -33,3 +33,53 @@ def test_failed_create_external_location(self):
...
+            "name": "regional site",
+            "address": "123 Test",
+            "country": "",
+            "location_type": "land_based",
+            "application_type": "sicl",
+        }
+
+        response = self.client.post(self.url, data, **self.exporter_headers)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```
[the remark this answers is not in the document]
Comment (rorysproule1) [reply organisations/tests/tests_external_locations.py:51]: Done!

File: organisations/tests/tests_external_locations.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rorysproule1) [reply organisations/tests/tests_external_locations.py:70]: Done!

File: applications/views/external_locations.py [hunk 6]
```diff
@@ -194,7 +201,11 @@ def _get_new_locations(
             for location_id in location_ids:
                 new_location = get_location(location_id, user_organisation)
 
-                if previous_location_countries and new_location.country.id not in previous_location_countries:
+                if (
+                    new_location.country
+                    and (previous_location_countries and new_location.country.id not in previous_location_countries)
+                    or new_location.location_type == "sea_based"
```
Comment (stephanieboyd) [inline applications/views/external_locations.py:207]: Is sea based not a constant somewhere? :)

Comment (rorysproule1) [reply applications/views/external_locations.py:207]: Good spot! Added

File: applications/views/external_locations.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply applications/views/external_locations.py:103]: For me it still says `if new_location.country and new_location.country.id`

File: organisations/tests/tests_external_locations.py [hunk 7]
```diff
@@ -33,3 +35,62 @@ def test_failed_create_external_location(self):
 
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
         self.assertEqual(ExternalLocation.objects.all().count(), 1)
+
```
Comment (BenAndrew-Kainos) [inline organisations/tests/tests_external_locations.py:38]: You're missing a test to check you get the `LOCATION_TYPE` error if you don't pass location type

**Pull request merged**